awx/main/tasks.py

```diff
@@ -1473,14 +1474,22 @@ def build_env(self, job, private_data_dir, isolated=False, private_data_files=No
             if authorize:
                 env['ANSIBLE_NET_AUTH_PASS'] = network_cred.get_input('authorize_password', default='')
 
-        for env_key, folder, default in (
-                ('ANSIBLE_COLLECTIONS_PATHS', 'requirements_collections', '~/.ansible/collections:/usr/share/ansible/collections'),
-                ('ANSIBLE_ROLES_PATH', 'requirements_roles', '~/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles')):
+        path_vars = (
+            ('ANSIBLE_COLLECTIONS_PATHS', 'collections_path', 'requirements_collections', '~/.ansible/collections:/usr/share/ansible/collections'),
+            ('ANSIBLE_ROLES_PATH', 'roles_path', 'requirements_roles', '~/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles'))
+
+        config_values = read_ansible_config(job.project.get_project_path(), list(map(lambda x: x[1], path_vars)))
```

Comment:
At this point, we've already set up the temporary directory and all the stuff in it. So if people wanted to avoid parsing the config as much as Earthly possible, we could avoid it in the event that the collection & role directories were empty.

Reply:
My only concern here would be issues parsing the file causing problems with the task executor requiring a restart. We might want to catch any exception that comes out of this and raise it as the `reason` a job failed.

Reply:
updated

Reply:
I don't think we want to `raise` it - if we fail to parse it, I'd just continue.